### DISCUSSION TITLE
I want to define my own function for positive electrode OCP

### DISCUSSION THREAD

**Discussion by kumaryash7** [pybamm-team/PyBaMM#2806] [Q&A]
I want to define a function for positive electrode OCP in which I have a graph between soc and positive equilibrium potential, where I  want to interpolate the equilibrium potential values according to the corresponding soc values from the graph. 
Is there any predefined function in pybamm for the same?

**Comment by valentinsulzer** [comment]
```python
def ocp(sto):
    return pybamm.Interpolant(x, y, sto, interpolator="cubic")

parameter_values["Negative electrode OCP [V]"] = ocp
```
where `x` are your soc values and `y` are your OCP values (both numpy arrays)

**Reply by kumaryash7** [reply]
I have the following as my soc values
array([0.40287624, 0.41599842, 0.4297446 , 0.44031858, 0.45194996,
       0.46463874, 0.47627013, 0.4921311 , 0.50481988, 0.51750866,
       0.53294668, 0.54478954, 0.5576898 , 0.57037858, 0.58200997,
       0.59639058, 0.60738753, 0.62060501, 0.63382249, 0.64915476,
       0.66131484, 0.6742151 , 0.6866924 , 0.70001562, 0.71453723,
       0.72793094, 0.74220582, 0.75838401, 0.77445647, 0.78820265,
       0.80237179, 0.8178098 , 0.83367078, 0.84530216, 0.85693354,
       0.86962233, 0.88231111, 0.89711468, 0.91297566, 0.92460704,
       0.93993932, 0.9520994 , 0.96478818, 0.97747696, 0.98593615,
       0.99122314, 0.99439533, 0.99651013, 0.99862493, 0.99862493,
       0.9933039 , 0.99587932, 0.99690949])


and the following as my OCP values 
array([4.2879116 , 4.25454591, 4.22392693, 4.19562757, 4.1714262 ,
       4.1478434 , 4.12232758, 4.09139932, 4.06839643, 4.04384712,
       4.01655293, 3.9966042 , 3.97541834, 3.95608817, 3.93907763,
       3.92005675, 3.90474726, 3.8892058 , 3.87412828, 3.85673113,
       3.84397322, 3.8319112 , 3.82077702, 3.8105707 , 3.79912724,
       3.79062197, 3.7825033 , 3.77252893, 3.76355974, 3.75640758,
       3.74933274, 3.74190995, 3.73379128, 3.72799223, 3.72180658,
       3.71593021, 3.70827547, 3.69899699, 3.6885587 , 3.67928022,
       3.66768212, 3.65469225, 3.63984669, 3.61618657, 3.58804185,
       3.54590209, 3.48211255, 3.41484357, 3.33133727, 3.29672217,
       3.51864407, 3.4519774 , 3.37740113])


But when I run my experiment I get this error:-
`x` must be strictly increasing sequence.

**Reply by valentinsulzer** [reply]
Your soc values are not strictly increasing
